build-tools/packages/build-cli/src/library/commands/buildProjectPackageCommand.ts

```diff
@@ -0,0 +1,269 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+	type IPackage,
+	type PackageFilterOptions,
+	type PackageSelectionCriteria,
+	type ReleaseGroupName,
+	selectAndFilterPackages,
+} from "@fluid-tools/build-infrastructure";
+import { type Command, Flags, ux } from "@oclif/core";
+import async from "async";
+
+import { BaseCommandWithBuildProject } from "./base.js";
+
+/**
+ * A re-usable CLI flag to parse release group names.
+ * Defined locally to avoid circular dependency with flags.ts.
+ */
+const releaseGroupNameFlag = Flags.custom<ReleaseGroupName>({
+	description: "The name of a release group.",
+	char: "g",
+	parse: async (input) => {
+		return input as ReleaseGroupName;
+	},
+});
+
+/**
+ * The default to use as selection criteria when none is explicitly provided by the user. This enables commands
+ * without flags to operate on a collection of packages by default that make sense based on the command.
+ */
+export type PackageSelectionDefault = "all" | "dir" | undefined;
```

Comment:
doesn't this type already exist in the current version of the comand? Avoid creating new types unless necessary.